ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -148,27 +334,132 @@ private[dao] final class TransactionsReader(
       .watchTermination()(endSpanOnTermination(span))
   }
 
+  sealed trait PointWiseTransactionFetching {
```

Comment:
it's ok for PoC but for the real thing I would definitely favor composition over inheritance here. this seems to me like simply a higher order function.

Reply:
For now I decided to stick with the inheritance approach